### ISSUE TITLE
zinject + scrub = deadlocks

### ISSUE THREAD

**Issue by ghost** [zfsonfreebsd/ZoF#91]
The `zpool_scrub_print_repairing` test gets stuck in the scrub command, and any command attempting to inspect the state of the pool hangs as well.

To reproduce:
```
truncate -s 10g disk0
zpool create testpool `pwd`/disk0
zinject -d `pwd`/disk0 -e io -T read -f 100 testpool
zinject -d `pwd`/disk0 -D10:1 testpool
zpool scrub testpool
```

**Comment by ghost** [comment]
I tested a few other commands before doing the scrub to try isolating whether scrub is of any significance. `zpool status` and `zfs diff` and `zfs list` and listing out some files do not hang, but `zpool scrub testpool` consistently does.

**Comment by mattmacy** [comment]
```
zpool create -f testpool mirror md0 md1
zinject -d md0 -e io -T read -f 100 testpool
zinject -d md0 -D10:1 testpool
zpool scrub testpool
```

**Comment by mattmacy** [comment]
fixed by dc8e75adafd4a72999209e990cc99689f5cdbd47
